src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.spec.ts

```diff
@@ -0,0 +1,249 @@
+    });
+
+    it('creates an instance of GooglePayPaymentStrategy', () => {
+        expect(strategy).toBeInstanceOf(CheckoutcomGooglePayPaymentStrategy);
+    });
+
+    describe('#execute', () => {
+        let googlePayOptions: PaymentInitializeOptions;
+
+        beforeEach(() => {
+            jest.spyOn(walletButton, 'addEventListener');
+            jest.spyOn(store, 'dispatch').mockReturnValue(Promise.resolve());
+            googlePayOptions = {
+                methodId: 'googlepaycheckoutcom',
+                googlepaycheckoutcom: {
+                    walletButton: 'mockButton',
+                    onError: () => {},
+                    onPaymentSelect: () => {},
+                },
+            };
+        });
+
+        it('creates the order and submit payment', async () => {
+
+            jest.spyOn(orderActionCreator, 'submitOrder').mockReturnValue(Promise.resolve());
+            jest.spyOn(paymentActionCreator, 'submitPayment').mockReturnValue(Promise.resolve());
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(googlePaymentMethodData);
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethodOrThrow').mockReturnValue(googlePaymentMethodData);
+
+            await strategy.initialize(googlePayOptions);
+            await strategy.execute(getGoogleOrderRequestBody());
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalled();
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalled();
+        });
+
+        it('throws and error when GooglePaymentOptions are not available', () => {
+            expect(() => strategy.execute(getGoogleOrderRequestBody())).toThrowError('Unable to initialize payment because "options.googlepaycheckoutcom" argument is not provided.');
+        });
+
+        it('gets again the payment information and submit payment', async () => {
+            const paymentData = getGooglePaymentDataMock();
+
+            jest.spyOn(orderActionCreator, 'submitOrder').mockReturnValue(Promise.resolve());
+            jest.spyOn(paymentActionCreator, 'submitPayment').mockReturnValue(Promise.resolve());
+            jest.spyOn(googlePayPaymentProcessor, 'handleSuccess').mockReturnValue(Promise.resolve());
+            jest.spyOn(googlePayPaymentProcessor, 'displayWallet').mockReturnValue(Promise.resolve(paymentData));
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(googlePaymentMethodData);
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethodOrThrow').mockReturnValue(googlePaymentMethodData);
+
+            await strategy.initialize(googlePayOptions);
+            await strategy.execute(getGoogleOrderRequestBody());
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalled();
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalled();
+        });
+
+        it('gets again the payment information and gets an error', async () => {
```

Comment:
since you are asserting that the payment information is retrieved again when an error is returned when displaying the wallet, might be worth having an assertion like
```js
expect(store.getState().paymentMethods, 'getPaymentMethod').toHaveBeenCalled();
```

The fact that you are mocking/spying on it, is not a guarantee that is being called.

In general, the test title should match the expectations in the body, e.g:

```js
describe('#method', () =>
  it('<assertions> [when <preconditions>]', () => {
    // .. preconditions, e.g: jest.spyOn().return(error)
    method();
   // ...assertions, eg: expect(...).toHaveBeenCalled()
  });
});
```

Reply:
fixed by expecting displayBallet to be called, as well as getPaymentMethod